Sources/App/Views/Home/HomeIndex+StructuredDataSchema.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+
+extension HomeIndex {
+    
+    struct IndexSchema: Encodable {
+        enum CodingKeys: String, CodingKey {
+            case context = "@context", type = "@type"
+            case url, potentialAction
+        }
+        
+        var context: String = "https://schema.org"
+        var type: String = "WebSite"
+        
+        let url: String
+        let potentialAction: SearchActionSchema
+        
+        init() {
+            self.url = SiteURL.home.absoluteURL()
+            self.potentialAction = .init(
+                target: .init(
+                    urlTemplate: SiteURL.search.absoluteURL(parameters: [
+                        .init(key: "query", value: "{search_term_string}")
```

Comment:
One thing I'm not 100% sure on is the fact that this parameters function will encode the `{` and `}` values - this is not what Google's example shows. It's plausible that our encoding here will actually prevent the whole feature from working.

We may want to try something more specific such as:
```swift
SiteURL.search.absoluteURL() + "?search={search_term_string}"
```
Or we could pass a new option through to `disableEncoding: true` or something which skips that step? This might be a bit nicer

Reply:
Is there a way for us to test this? I'm not sure there is. We block Google from indexing staging.

It's not the end of the world if we have to test it on prod, it's not like it'll break anything on the site itself.

Reply:
The other issue is we don't know how long it takes for Google to reindex and enable this feature even if it were to work 😅

Reply:
Can we infer it from looking at other sites that implement this? 🤔

Reply:
If we're to infer, I feel like we have to follow Google's own documentation which has it unencoded. Would you like me to go down the `disableEncoding` idea and get us as close to the docs as possible?

Reply:
I tried looking for another site that supported it and found https://www.asos.com/ does.

However, it does not appear to implement the `SearchAction` structured data. I wonder if they're doing it a different way?

Reply:
Google definitely have some intelligence they use to capture most website's search features without explicit support for the structured data (either form) - generally for a lot of big websites that magically fit their algorithm!

Reply:
Let's go as close to the docs as possible then 🚀

Reply:
Have implemented my solution, let me know if you'd rather tackle it in a different way.